vcloud/box/common/bootstrap.erb

```diff
@@ -1,26 +1,31 @@
 #!/bin/bash
 
 if [ $1 = "postcustomization" ]; then
-  echo 'ubuntu:<%= vars[:pass_hash] %>' | chpasswd -e 
-  
+  echo 'ubuntu:<%= vars[:pass_hash] %>' | chpasswd -e
+
   dpkg-reconfigure openssh-server
   set -e
   exec >/var/log/userdata.log 2>&1
   export DEBIAN_FRONTEND=noninteractive
   E="/etc/environment"
   echo "LC_ALL=en_GB.UTF-8" > $E
-  
-  R="/etc/resolv.conf"
-  echo "nameserver 8.8.8.8" >$R
-  echo "nameserver 8.8.4.4" >>$R
-  
+
+  <% if vars[:dns_suffix] %>
+    sed -i '/^127.0.1.1/d' /etc/hosts
+    sed -i '/^dns-search/d' /etc/network/interfaces
+    sed -i '/^dns-nameservers/d' /etc/network/interfaces
+    echo -e 'dns-search\t<%= vars[:dns_suffix] %>\ndns-nameservers\t8.8.8.8 8.8.4.4' >> /etc/network/interfaces
+    echo -e 'nameserver 8.8.8.8\nnameserver 8.8.4.4\nsearch <%= vars[:dns_suffix] %>' | resolvconf -a eth0.inet
+  <% end %>
+
```

Comment:
I don't think we should merge this until it's clear what changes we make - otherwise we create more work for people looking at this in future who will spend time trying to understand why these choices were made. What you describe doesn't match what we've seen in our other vCloud environments.

I'm happy to pair on this to do further testing.